include/seqan3/argument_parser/validators.hpp

```diff
@@ -297,68 +290,347 @@ class file_ext_validator
     //!\endcond
     void operator()(range_type const & v) const
     {
-        std::for_each(v.begin(), v.end(), [&] (auto cmp) { (*this)(cmp); });
+         std::for_each(v.begin(), v.end(), [&] (auto cmp) { this->operator()(cmp); });
+    }
+
+protected:
+
+    /*!\brief Validates the given filename path based on the specified extensions.
+     * \param path The filename path.
+     * \throws parser_invalid_argument if the specified extensions don't match the given path.
+     */
+    void validate_path_impl(std::filesystem::path const & path) const
```

Comment:
I know this is picky, but for me a path is either a directory or a filename. In this function you except a filename so maybe it should be called validate_filename_impl?

Reply:
Yes, this is picky 🤣 . But honestly, I really don't see the point of changing this, as it is not even a public interface but a protected used internally.

Reply:
it is still inconsistent with path_is_readable which acceppts both, file and dir.. :(
How picky am I allowed to be? (I just think it is confusing for the person who looks at it a ywar from now)

Reply:
> How picky am I allowed to be?

Well, no one forbids you to be picky and it is perfectly fine to review this as critical as possible. But that doesn't mean that I have to agree with it 😇. And I still don't see the point, though I have changed it already.